packages/relay/src/lib/eth.ts

```diff
@@ -571,7 +571,34 @@ export class EthImpl implements Eth {
 
               // The block is from the last 15 minutes, therefore the historical balance hasn't been imported in the Mirror Node yet
               if (timeDiff < constants.BALANCES_UPDATE_INTERVAL) {
-                throw predefined.UNKNOWN_HISTORICAL_BALANCE;
+
+                const balance = await this.mirrorNodeClient.getBalanceAtTimestamp(mirrorAccount.account, undefined, requestId);
+                let currentBalance = 0;
+                let currentTimestamp;
+                let balanceFromTxs = 0;
+
+                if (balance.balances?.length) {
+                  currentBalance = balance.balances[0].balance;
+                  currentTimestamp = balance.timestamp;
+                }
+
+                let transactionsInTimeWindow = await this.mirrorNodeClient.getTransactionsForAccount(
+                    mirrorAccount.account,
+                    block.timestamp.from,
```

Comment:
Q: should this be `from` or `to`?
Just checking we don't replay the transaction within a block if the retrieved balance already includes that.
We should verify this in an E2E test.

Reply:
After some further consideration I changed it to `to`, since it makes more sence. When requesting the balance at block X, all transactions from within that block should be accounted for in the calculated balance, hence we should use `to` in this case.